sql/fulltext/fulltext.go

```diff
@@ -0,0 +1,687 @@
+		return err
+	}
+	return editor.Insert(ctx, new)
+}
+
+// Delete implements the interface sql.TableEditor.
+func (editor Editor) Delete(ctx *sql.Context, row sql.Row) error {
+	for _, index := range editor.Indexes {
+		// Grab the source columns
+		//TODO: probably should cache this or something
+		sourceCols := make([]interface{}, len(index.SourceCols))
+		for i, sourceCol := range index.SourceCols {
+			sourceCols[i] = row[sourceCol]
+		}
+
+		//TODO: support keyless, this assumes primary or unique
+		refCols := make([]interface{}, len(index.RefCols.Positions))
+		for i, refCol := range index.RefCols.Positions {
+			refCols[i] = row[refCol]
+		}
+
+		// Create a parser for the source columns, then iterate over the words
+		parser := NewDefaultParser(ctx, sourceCols...)
+		word, position, reachedTheEnd, err := parser.Next(ctx)
+		for ; err == nil && !reachedTheEnd; word, position, reachedTheEnd, err = parser.Next(ctx) {
+			// Delete from the position table
+			//TODO: probably should cache this or something
+			positionRow := make(sql.Row, 1, len(SchemaPosition)+len(refCols))
+			positionRow[0] = word
+			positionRow = append(positionRow, refCols...)
+			positionRow = append(positionRow, position)
+			if err = index.Position.Delete(ctx, positionRow); err != nil {
+				return err
+			}
+
+			// Delete from the document count table
+			docCountRow := make(sql.Row, 1, len(SchemaDocCount)+len(refCols))
+			docCountRow[0] = word
+			docCountRow = append(docCountRow, refCols...)
+			docCountRow = append(docCountRow, uint64(0)) // Since count isn't in the PK, this should be safe (caught by tests if not)
+			if err = index.DocCount.Delete(ctx, docCountRow); err != nil {
+				return err
+			}
+			//TODO: delete from the remaining tables
+		}
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Close implements the interface sql.TableEditor.
+func (editor Editor) Close(ctx *sql.Context) error {
+	err := editor.Config.Close(ctx)
+	for _, editorIndex := range editor.Indexes {
+		if nErr := editorIndex.Position.Close(ctx); err == nil {
+			err = nErr
+		}
+		if nErr := editorIndex.DocCount.Close(ctx); err == nil {
+			err = nErr
+		}
+		if nErr := editorIndex.GlobalCount.Close(ctx); err == nil {
+			err = nErr
+		}
+	}
+	return err
+}
+
+// MultiTableEditor wraps multiple table editors, allowing for a single function to handle writes across multiple tables.
+type MultiTableEditor struct {
+	primary     sql.TableEditor
+	secondaries []sql.TableEditor
+}
+
+var _ sql.TableEditor = MultiTableEditor{}
+var _ sql.ForeignKeyEditor = MultiTableEditor{}
+var _ sql.AutoIncrementSetter = MultiTableEditor{}
+
+// CreateMultiTableEditor creates a TableEditor that writes to both the primary and secondary editors. The primary
+// editor must implement ForeignKeyEditor and AutoIncrementSetter in addition to TableEditor.
+func CreateMultiTableEditor(ctx *sql.Context, primary sql.TableEditor, secondaries ...sql.TableEditor) (sql.TableEditor, error) {
```

Comment:
the API for exposing a single FT editor might be a bit easier to use, not sure if implementation is more confusion